src/cpp/desktop/DesktopWebView.cpp

```diff
@@ -35,6 +37,46 @@
 namespace rstudio {
 namespace desktop {
 
+namespace {
+
+#if QT_VERSION >= QT_VERSION_CHECK(5, 11, 0)
+
+class DevToolsWindow : public QMainWindow
+{
+public:
+   
+   DevToolsWindow()
+      : webPage_(new QWebEnginePage(this)),
+        webView_(new QWebEngineView(this))
+   {
+      webView_->setPage(webPage_);
+      
+      QScreen* screen = QApplication::primaryScreen();
+      QRect geometry = screen->geometry();
+      resize(geometry.width() * 0.7, geometry.height() * 0.7);
+      
+      setWindowTitle(QStringLiteral("RStudio DevTools"));
+      setAttribute(Qt::WA_DeleteOnClose, false);
+      setAttribute(Qt::WA_QuitOnClose, true);
+      
+      setUnifiedTitleAndToolBarOnMac(true);
+      setCentralWidget(webView_);
+   }
+   
+   QWebEnginePage* webPage() { return webPage_; }
+   QWebEngineView* webView() { return webView_; }
+   
+private:
+   QWebEnginePage* webPage_;
+   QWebEngineView* webView_;
+};
+
+std::map<QWebEnginePage*, DevToolsWindow*> s_devToolsWindows;
```

Comment:
Note to self: I should really just make the DevToolsWindow a member variable of the WebPage, and associate its lifetime with that of the associated page (so that if the page goes away the devtools page does too). As it stands now we're leaking a small map of pointers as RStudio is run (although that's probably not a leak worth worrying about?)

Reply:
I don't think leaking that map is worth worrying about. Do you want to make the member change before this is merged? (Or just at some point in the future?)